batavia/types/Complex.js

```diff
@@ -197,28 +198,151 @@ Complex.prototype.__abs__ = function() {
 /**************************************************
  * Binary operators
  **************************************************/
-
-Complex.prototype.__pow__ = function(exponent) {
+function hyp(x, y) {
+    x = Math.abs(x)
+    y = Math.abs(y)
+    if (x < y) {
+        var temp = x
+        x = y
+        y = temp
+    }
+    if (x === 0) {
+        return 0
+    } else {
+        var yx = y / x
+        return x * Math.sqrt(1 + yx * yx)
+    }
+}
+function quot(a, b) {
+    var r = new Complex(0, 0)
+    var abs_bimag = 0
+    var abs_breal = 0
+    if (b.real < 0) {
+        abs_breal = -b.real
+    } else {
+        abs_breal = b.real
+    }
+    if (b.imag < 0) {
+        abs_bimag = -b.imag
+    } else {
+        abs_bimag = b.imag
+    }
+    if (abs_breal >= abs_bimag) {
+        if (abs_breal === 0) {
+            r = new Complex(0, 0)
+        } else {
+            const ratio = b.imag / b.real
+            const denom = b.real + b.imag * ratio
+            r = new Complex((a.real + a.imag * ratio) / denom, (a.imag - a.real * ratio) / denom)
+        }
+    } else if (abs_bimag >= abs_breal) {
+        const ratio = b.real / b.imag
+        const denom = b.real * ratio + b.imag
+        r = new Complex((a.real * ratio + a.imag) / denom, (a.imag * ratio - a.real) / denom)
+    } else {
+        r = new Complex(NaN, NaN)
+    }
+    return r
+}
+function powu(x, y) {
```

Comment:
This looks like it only works for positive integer `y`, but below it could be called with a positive real number?